Maki/added_db.py

```diff
@@ -0,0 +1,349 @@
+from os import path
+from bs4 import BeautifulSoup
+import requests
+import memcache
+from datetime import datetime
+import sys
+from http.server import HTTPServer, BaseHTTPRequestHandler
+import json
+import sqlite3 as sql
+
+ATTRIBUTE_NAMES = 'Date,Total_Cases,New_Cases,Total_Deaths,New_Deaths,Total_Recovered,Active_cases,' \
+                  'Serious_or_Critical,' \
+                  'Total_Cases_per_1M '
+
+ATTRIBUTE_NAMES_DB = 'Date,Total_Cases text,New_Cases integer,Total_Deaths integer,New_Deaths integer,Total_Recovered integer,Active_cases integer,' \
+                     'Serious_or_Critical integer,' \
+                     'Total_Cases_per_1M real'
+
+URL = "https://www.worldometers.info/coronavirus/?fbclid=IwAR1OutjUurc_K" \
+      "4BH9F4smkLpC0yKfndoShfUtrs4cJZehqS7PQs0Ek85Xlw"
+
+LOCAL_SERVER_IP = "127.0.0.1"
+PUBLIC_SERVER_IP = "0.0.0.0"
+PORT_NUMBER = 2021
+CSV_FILE_NAME = "stats_covid19.csv"
+DATABASE_NAME = "covid19.db"
+COVID_TABLE = "covid_19"
+
+STATS_DB_KEY = "stats_db"
+STATS_KEY = "stats"
+DATE_KEY = "last_datetime_key"
+HEADER_KEY = 'key_header'
+DATABASE_KEY = "db_key1"
+
+CACHE = memcache.Client(LOCAL_SERVER_IP)
+
+my_db = None
+
+LAST_UPDATED_DATE = CACHE.get(DATE_KEY)
+
+
+def main():
+    global LAST_UPDATED_DATE
+
+    if LAST_UPDATED_DATE is None:
+        LAST_UPDATED_DATE = set_date_to_cache()
+
+    if len(sys.argv) > 1:
+        if sys.argv[1] == '--https':
+            run()
+
+    # if --https is sent as argument the program stops here
+    update()
+
+
+def update_database():
+    global my_db
+    if my_db is None:
+        my_db = Database()
+        my_db.create_table(COVID_TABLE, ATTRIBUTE_NAMES_DB)
+        my_db.update(get_stats(db=True), COVID_TABLE)
+    if not my_db.is_up_to_date():
+        stats = get_stats()
+        my_db.update(stats, COVID_TABLE)
+
+
+class Database:
+    # how to make Database inherit and also put new fields?
+    def __init__(self):
+        self.tables = dict()
+        self.last_row = None
+        self.connection = sql.connect(DATABASE_NAME)
+        self.cursor = self.connection.cursor()
+
+    def execute(self, command, values=None):
+        print(command)
+        if values is None:
+            self.cursor.execute(command)
+            return
+        self.cursor.execute(command, values)
+
+    def commit(self):
+        self.connection.commit()
+
```

Comment:
Methods that are here to be used mainly by the other methods in the class are good to be private (with name starting with `__` _dpuble underline_)